### PR TITLE
fix JALR, and add new SRAM test

### LINKED ISSUE #32: JALR not working
### 1. Steps to reproduce the behavior: 

https://github.com/shuosc/shuorv/blob/f477bd4f7b4fd9e13f51746f10b7f9adc8df2adf/src/main/scala/Imm.scala#L14
check this JALR is I-Type instruction, should gen imm.

### 2. What did you expect to see? 
should be
```
 is("b11001".U) {
      io.result := I
    }
```
?

### PR DESCRIPTION
**Pull request by zhoudian64**
### What problem does this PR solve?

Issue Number: close #32

Problem Summary:

What's Changed: 
- fix JALR imm gen stage
- add new Top test, and find some RegFile bugs.

How it Works:
```
sw x0, 0(0x0FB8)
lw x14, 0(0x0FB8)
```
x14 == 0x80000086

### CODE REVIEW

**Comment by zhoudian64** [comment]
@longfangsong  take a look.

File: src/main/scala/Top.scala [hunk 1]
```diff
@@ -3,7 +3,9 @@ import chisel3._
 class Top extends Module {
   val io = IO(new Bundle {
     val gpio = Output(UInt(32.W))
+    val reset = Input(Bool())
```
Comment (longfangsong) [inline src/main/scala/Top.scala:6]: IMO this is an FPGA specific behaviour, we should not include this in source tree.

File: src/main/scala/ProgramROM.scala [hunk 2]
```diff
@@ -10,32 +10,156 @@ class ProgramROMBundle extends Bundle {
...
-    "h01c38eb3".U(32.W), // add t4, t2, t3
-    "h007eeeb3".U(32.W), // or t4, t4, t2
-    "h10012e37".U(32.W), // li t3, 0x10012000
-    "h000e0e13".U(32.W),
-    "h01de2023".U(32.W), // sw t4, 0(t3)
-    "h00138393".U(32.W), // addi t2, t2, 1
-    // j label
-    "hfe1ff06f".U(32.W)))
+//<_start>:
+"h00021197".U(32.W),    //auipc x3,0x21
```
Comment (longfangsong) [inline src/main/scala/ProgramROM.scala:14]: The new test is SO LARGE, which made it hard to understand.
But IMO we should not do intergration test in this f**king way anyhow, we can keep it like this for now, and find a more data-driven way to do it.

**Review by longfangsong — COMMENTED** [review]
Rest LGTM

**Review by longfangsong — APPROVED** [review]

**Comment by longfangsong** [comment]
I merged the bug fix part.

Let's find a better way to do tests.

**Comment by zhoudian64** [comment]
I want to remove TopTest.

**Pull request merged**